src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement statement = connection
+                    .prepareStatement(FIND_BY_ID_QUERY)) {
+                statement.setLong(1, id);
+                try (ResultSet resultSet = statement.executeQuery()) {
+                    if (resultSet.next()) {
+                        return Optional.of(mapToBook(resultSet));
+                    }
+                }
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot find book by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement statement = connection.prepareStatement(FIND_ALL_QUERY)) {
+                try (ResultSet resultSet = statement.executeQuery()) {
+                    List<Book> books = new ArrayList<>();
+                    while (resultSet.next()) {
+                        books.add(mapToBook(resultSet));
+                    }
+                    return books;
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Retrieving all books from DB was failed", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement statement = connection.prepareStatement(UPDATE_QUERY)) {
+                statement.setString(1, book.getTitle());
+                statement.setBigDecimal(2, book.getPrice());
+                statement.setLong(3, book.getId());
+                if (statement.executeUpdate() < 1) {
+                    throw new DataProcessingException(
+                            String.format("Book with id=%d wasn't updated, no rows affected.",
+                                    book.getId()));
+                }
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(
+                    String.format("Updating book with id=%d was failed", book.getId()), e
+            );
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement statement = connection.prepareStatement(DELETE_QUERY)) {
+                statement.setLong(1, id);
+                return statement.executeUpdate() > 0;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Deletion book by id=%d was failed", id), e
+            );
+        }
+    }
+
+    private Book mapToBook(ResultSet resultSet) throws SQLException {
+        return new Book(
+                resultSet.getLong("id"),
```

Comment:
look again at the [checklist](https://github.com/mate-academy/jv-jdbc-intro/blob/main/checklist.md)